### PR TITLE
5958 openssl wrapper class

### PR DESCRIPTION
**Pull request by konangc87**
|Related issue|
|---|
|5958|
# OpenSSL helper to hash data.

## Description
This issue aims to create a helper class to wrap open ssl APIs to hash data.

## DoD / Tasks
- [X] Implementation
- [X] RTR
image

image

### CODE REVIEW

File: src/shared_modules/utils/hashHelper.h [hunk 1]
```diff
@@ -0,0 +1,120 @@
+/*
+ * Wazuh DBSYNC
```
Comment (tinachoB) [inline src/shared_modules/utils/hashHelper.h:2]: Wazuh shared modules utils?

File: src/shared_modules/utils/hashHelper.h [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+ * Copyright (C) 2015-2020, Wazuh Inc.
+ * June 11, 2020.
```
Comment (tinachoB) [inline src/shared_modules/utils/hashHelper.h:4]: Please, fix date

File: src/shared_modules/utils/tests/hashHelper_test.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Wazuh DBSYNC
```
Comment (tinachoB) [inline src/shared_modules/utils/tests/hashHelper_test.cpp:2]: Wazuh shared modules utils?

File: src/shared_modules/utils/tests/hashHelper_test.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Wazuh DBSYNC
```
Comment (tinachoB) [inline src/shared_modules/utils/tests/hashHelper_test.h:2]: Wazuh shared modules utils?

File: src/shared_modules/utils/hashHelper.h [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            };
+            // LCOV_EXCL_START
+            if (!ret)
+            {
+                throw std::runtime_error
+                {
+                    "Error getting digest final."
+                };
+            }
+            // LCOV_EXCL_START
```
Comment (tinachoB) [inline src/shared_modules/utils/hashHelper.h:50]: LCOV_EXCL_STOP

File: src/shared_modules/utils/hashHelper.h [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                throw std::runtime_error
+                {
+                    "Error getting digest final."
+                };
+            }
+            // LCOV_EXCL_STOP
+            return {digest, digest+digestSize};
+        }
+    private:
+        struct EvpContextDeleter
```
Comment (tinachoB) [inline src/shared_modules/utils/hashHelper.h:72]: please, add final here

**Review by tinachoB — CHANGES_REQUESTED** [review]

**Review by tinachoB — APPROVED** [review]
Great work

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**